include/visitors/code_gen.h

```diff
@@ -2,30 +2,545 @@
+        this->environment->push_env();
+
+        for (auto &stmt : *stmts)
+        {
+            if (auto func_stmt = dynamic_cast<Func *>(stmt.get()))
+            {
+                func_stmt->accept(this);
+            }
+
+            if (auto decl_stmt = dynamic_cast<DeclStmt *>(stmt.get()))
+            {
+                decl_stmt->accept(this);
+            }
+
+            if (auto print_stmt = dynamic_cast<PrintStmt *>(stmt.get()))
+            {
+                print_stmt->accept(this);
+            }
+
+            if (auto if_stmt = dynamic_cast<IfStmt *>(stmt.get()))
+            {
+                if_stmt->accept(this);
+            }
+
+            if (auto block = dynamic_cast<Block *>(stmt.get()))
+            {
+                block->accept(this);
+            }
+
+            if (auto expr_stmt = dynamic_cast<ExprStmt *>(stmt.get()))
+            {
+                expr_stmt->accept(this);
+            }
+
+            if (auto ternary_stmt = dynamic_cast<Ternary *>(stmt.get()))
+            {
+                ternary_stmt->accept(this);
+            }
+
+            if (auto while_stmt = dynamic_cast<WhileStmt *>(stmt.get()))
+            {
+                while_stmt->accept(this);
+            }
+
+            if (auto for_stmt = dynamic_cast<ForStmt *>(stmt.get()))
+            {
+                for_stmt->accept(this);
+            }
+
+            if (auto return_stmt = dynamic_cast<ReturnStmt *>(stmt.get()))
+            {
+                return_stmt->accept(this);
+            }
+
+            if (auto break_stmt = dynamic_cast<BreakStmt *>(stmt.get()))
+            {
+                break_stmt->accept(this);
+            }
+
+            if (auto continue_stmt = dynamic_cast<ContinueStmt *>(stmt.get()))
+            {
+                continue_stmt->accept(this);
+            }
+        }
+
+        create_entry_point();
+
+        BinaryenModulePrint(this->mod);
+
+        BinaryenModuleAllocateAndWriteResult result =
+            BinaryenModuleAllocateAndWrite(this->mod, nullptr);
+
+        if (!result.binary || result.binaryBytes == 0)
+        {
+            std::cerr << "failed to serialize" << std::endl;
+            return;
+        }
+
+        std::string filename = "output.wasm";
+        std::ofstream file(filename, std::ios::binary);
+        if (file.is_open())
+        {
+            file.write(static_cast<char *>(result.binary), result.binaryBytes);
+            file.close();
+            std::cout << "wasm module written to " << filename << std::endl;
+        }
+        else
+        {
+            std::cerr << "failed to open file" << filename << std::endl;
+        }
+
+        free(result.binary);
+
+        this->environment->pop_env();
+    }
+
+    int allocate_local()
+    {
+        int index = this->local_index++;
+        return index;
+    }
+
+    BinaryenType from_bird_type(Token token)
+    {
+        if (token.lexeme == "bool")
+            return BinaryenTypeInt32(); // no i1?
```

Comment:
The only types are `i32`, `i64`, `f32`, and `f64`. You've got it right in the code here. Bools are represented by `i32`.